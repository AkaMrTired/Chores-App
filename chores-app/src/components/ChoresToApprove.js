import React from "react";
import PropTypes from "prop-types";
import ChoreCard from "./ChoreCard";

const ChoresToApprove = ({ chores }) => {
  return (
    <div className="container">
      <h2>These are the completed chores for you to approve</h2>
      <p>
        When you accept a chore is done, the child will get the reward in their
        account
      </p>
      <p>If you click reject, it will be added back onto their todo list</p>

      {/* mapping function to go through the chores list and render the pending chores, it would be nice to add who is doing the chore in the future */}
      {chores
        .filter((chore) => chore.status === "P")
        .map((chore) => (
          <ChoreCard
            key={chore._id}
            name={chore.name}
            price={chore.price}
            component="ChoresToApprove"
          />
        ))}
      <div>
        <button type="button">
          <a href="/parentdashboard">Back To Dashboard</a>
        </button>
      </div>
    </div>
  );
};

// we will need to edit the prop validation when we know what the data looks like.

ChoresToApprove.propTypes = {
  chores: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.number,
      name: PropTypes.string,
      price: PropTypes.number,
      status: PropTypes.string,
    })
  ).isRequired,
};
export default ChoresToApprove;
