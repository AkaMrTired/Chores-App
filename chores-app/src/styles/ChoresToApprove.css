.approve-chores-container {
    max-width: var(--mobile-width);
    margin: 2rem auto;
    background: var(--gradient);
    padding: 2rem;
    border-radius: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
}
.approve-chores-container span {
    margin-bottom: 1rem;
}
.approve-chores-container span:last-of-type {
    margin-bottom: 2rem;
}