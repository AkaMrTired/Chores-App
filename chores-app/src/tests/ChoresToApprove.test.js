import { render, screen } from "@testing-library/react";
import React from "react";
import ChoresToApprove from "../components/ChoresToApprove";

const validChores = [
  { _id: 123, name: "takenChore", price: "1", status: "T" },
  { _id: 1234, name: "availableChore", price: "1", status: "A" },
  { _id: 12345, name: "unavailableChore", price: "1", status: "U" },
  { _id: 123456, name: "pendingApprovalChore", price: "1", status: "P" },
];

describe("the parents dashboard", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<ChoresToApprove chores={validChores} />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render basic fields", () => {
    render(<ChoresToApprove chores={validChores} />);
    expect(
      screen.getByText(/these are the completed chores/i)
    ).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /accept/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /reject/i })).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /back to dashboard/i })
    ).toBeInTheDocument();
  });
  it("should render only the chores with the available status", () => {
    render(<ChoresToApprove chores={validChores} />);
    expect(screen.queryByText(/availableChore/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/unavailableChore/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/takenChore/i)).not.toBeInTheDocument();
    expect(screen.getByText(/pendingApprovalChore/i)).toBeInTheDocument();
  });
});
