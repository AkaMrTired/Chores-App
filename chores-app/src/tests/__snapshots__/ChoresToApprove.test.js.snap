// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the parents dashboard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      These are the completed chores for you to approve
    </h2>
    <p>
      When you accept a chore is done, the child will get the reward in their account
    </p>
    <p>
      If you click reject, it will be added back onto their todo list
    </p>
    <div>
      <div>
        pendingApprovalChore
      </div>
      <div>
        Â£1
      </div>
      <button
        type="button"
      >
        Accept
      </button>
      <button
        type="button"
      >
        Reject
      </button>
    </div>
    <div>
      <button
        type="button"
      >
        <a
          href="/parentdashboard"
        >
          Back To Dashboard
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
